airflow/jobs/scheduler_job_runner.py

```diff
@@ -1083,8 +1084,13 @@ def _do_scheduling(self, session: Session) -> int:
             callback_tuples = self._schedule_all_dag_runs(guard, dag_runs, session)
 
         # Send the callbacks after we commit to ensure the context is up to date when it gets run
+        # cache saves time during scheduling of many dag_runs for same dag
+        cached_dags: dict = {}
         for dag_run, callback_to_run in callback_tuples:
-            dag = self.dagbag.get_dag(dag_run.dag_id, session=session)
+            dag = get_value_with_cache(
+                cached_dags, dag_run.dag_id, self.dagbag.get_dag, dag_run.dag_id, session=session
+            )
```

Comment:
Passing the params individually is a convoluted way to do it... Using a lambda you can just pass the whole packaged call as a Callable without the need for extra parameters
```suggestion
            dag = get_value_with_cache(
                cached_dags, dag_run.dag_id, lambda: self.dagbag.get_dag(dag_run.dag_id, session=session)
            )
```

Reply:
Thanks for the feedback! Using now lambda.